Updated Light Theme Color Token descriptions to the latest

Akatori-Design opened the pull request:
Closes: https://github.com/MetaMask/design-tokens/issues/59

Updated Light Theme color token descriptions to reflect our latest usage practice guide. This is so we can help designers to apply color tokens correctly following our practices when building new components.

* Colors we no longer wish to support are prefixed with "[Deprecated]" with the old description in tact.
* HEX values are NOT subject to the updates for this PR. They all remain the same.
* Once review is done, I'll make the same update on the Dark Theme.

The latest descriptions are also viewable in the Figma file
https://www.figma.com/file/kdFzEC7xzSNw7cXteqgzDW/%5BColor%5D-Light-Theme?node-id=0%3A1

georgewrmarshall reviewed (changes requested):
Most of the changes make sense to me. Just have one question about the secondary color and what we should update it's current usage too if we are fully deprecating it?

Also could you update the dark theme color descriptions as well?

georgewrmarshall commented on src/figma/tokens.json:
```diff
@@ -329,190 +329,190 @@
           "value": "#F66A0A",
-          "description": "(orange500: #F66A0A) secondary.default should be used for any secondary actions. It should not be used for any negative connotations such as warnings or errors as it is quite closely tied to the MetaMask Fox",
+          "description": "(orange500: #F66A0A) [Deprecated] secondary.default should be used for any secondary actions. It should not be used for any negative connotations such as warnings or errors as it is quite closely tied to the MetaMask Fox",
           "type": "color"
         },
         "alternative": {
           "value": "#C65507",
-          "description": "(orange600: #C65507) secondary.alternative should be used as an alternative to secondary.default for things such as hover states",
+          "description": "(orange600: #C65507) [Deprecated] secondary.alternative should be used as an alternative to secondary.default for things such as hover states",
           "type": "color"
         },
         "muted": {
           "value": "#F66A0A19",
-          "description": "(orange500: #F66A0A 10% opacity) secondary.muted is a very low contrasting secondary variant for things such as alert backgrounds. secondary.muted and secondary.inverse should not be used together in a foreground and background combination",
+          "description": "(orange500: #F66A0A 10% opacity) [Deprecated] secondary.muted is a very low contrasting secondary variant for things such as alert backgrounds. secondary.muted and secondary.inverse should not be used together in a foreground and background combination",
           "type": "color"
         },
         "inverse": {
           "value": "#FCFCFC",
-          "description": "(white010: #FCFCFC) secondary.inverse should be used only as the foreground element on top of secondary.default and secondary.alternative. It is intended to be the most contrasting color to secondary.default. It should meet all AA and AAA accessibility standards such as the text or icon of a secondary button.",
+          "description": "(white010: #FCFCFC) [Deprecated] secondary.inverse should be used only as the foreground element on top of secondary.default and secondary.alternative. It is intended to be the most contrasting color to secondary.default. It should meet all AA and AAA accessibility standards such as the text or icon of a secondary button.",
           "type": "color"
         },
         "disabled": {
           "value": "#F66A0A80",
-          "description": "(orange500: #F66A0A 50% opacity) secondary.disabled should be used for all disabled secondary action components",
+          "description": "(orange500: #F66A0A 50% opacity) [Deprecated] secondary.disabled should be used for all disabled secondary action components",
           "type": "color"
         }
       },
```
What are we suggesting we update the current secondary color use case to?

<img width="359" alt="Screen Shot 2022-04-08 at 1 16 31 PM" src="https://user-images.githubusercontent.com/8112138/162525501-004e6017-c793-422b-9a3f-05346f8caefb.png">

![Screen Shot 2022-04-08 at 1 17 42 PM](https://user-images.githubusercontent.com/8112138/162525503-fc581322-0aca-45f6-91be-76891bc956de.png)

![Screen Shot 2022-04-08 at 1 17 50 PM](https://user-images.githubusercontent.com/8112138/162525506-aee21d84-f835-42c9-818d-801f4046c87b.png)

<img width="341" alt="Screen Shot 2022-04-08 at 1 19 12 PM" src="https://user-images.githubusercontent.com/8112138/162525507-6d8bbe7b-aebb-439c-bdc8-5265e37bad14.png">

georgewrmarshall replied:
Resolved in internal chat.

georgewrmarshall reviewed (commented):
Secondary color comment

Akatori-Design commented:
I think we can keep the orange in our color, but it can't be in both `Secondary` & `Warning`. I suggest we deprecate `Secondary` palette, then add orange to the `Warning` palette so it justifies existing use cases. Any use cases for `Secondary` semantic that exist ( like the preloader) should be normalized to our updated standards, so for example replace prelaoder color with `icon.default`.

Akatori-Design commented:
Just updated all Dark theme descriptions to match the light theme ones. Only descriptions. #hex unchanged
Also fixed few inverse descriptions in the Light theme that had incorrect descriptions mapped.

georgewrmarshall reviewed (approved):
lgtm

georgewrmarshall commented on src/figma/tokens.json:
```diff
@@ -329,232 +329,232 @@
         "default": {
           "value": "#037DD6",
-          "description": "(blue500: #037DD6) primary.default should be used for all primary action components such as buttons or links.",
+          "description": "(blue500: #037DD6) For interactive or active semantic",
```
@Akatori-Design If we are using this for all active semantics should the check icon when and account is active be blue? Also there are many "interactive" components that don't use blue? icon/default, danger/critical buttons etc

Akatori-Design replied:
Sorry, missed this comment.  My logic was,  for colors with clear purpose I stated  "For (purpose)".  For color options  relative to the default options (stronger, weaker...etc), I stated "(adjective) option". Were the descriptions confusing you?

Akatori-Design replied:
- You're right about the active state. It should be blue if it's communicating "selected" state, not green cuz green is like success or positive. 
- Yea Im not exactly sure on the primary description. How would you describe it? "For primary user action related elements"?

georgewrmarshall replied:
Yeah I think "For primary user action related elements" is more specific. I can make some suggestions and you can commit if you like.

Akatori-Design replied:
Sounds good.

georgewrmarshall commented on src/figma/tokens.json:
```diff
@@ -329,232 +329,232 @@
         "alternative": {
           "value": "#0260A4",
-          "description": "(blue600: #0260A4) primary.alternative should be used as an alternative to primary.default for things such as hover states",
+          "description": "(blue600: #0260A4) Reserved for \"pressed\" state on interactive elements.",
```
Again should we suggest primary elements here?
```suggestion
          "description": "(blue600: #0260A4) Reserved for \"pressed\" state on primary elements.",
```

georgewrmarshall reviewed (changes requested):
Should we try to make our language more consistent? At the beginning of some descriptions we use "For" and some times its omitted? Especially if it's in the same sub category. e.g. All "muted" descriptions
